src/extension/openFileAtLocation.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {ExtensionMessage, MessagingChannel} from "../common/extensionMessaging";
+import {InterProcessMessageSender} from "../common/interProcessMessageSender";
+import {ReactNativeProjectHelper} from "../common/reactNativeProjectHelper";
+import * as path from "path";
+
+/* Usage:
+...path\openFileAtLocation.js filename:lineNumber
+...path\openFileAtLocation.js filename
+*/
+
+try {
+    const fileInfo = process.argv[2].split(":");
+    const filename = fileInfo[0];
+    let lineNumber = "";
+    if (fileInfo.length >= 2) lineNumber = fileInfo[1];
+
+    getReactNativeWorkspaceForFile(filename).then(projectRootPath => {
+        const remoteExtensionServerPath = new MessagingChannel(projectRootPath).getPath();
+        const interProcessMessageSender = new InterProcessMessageSender(remoteExtensionServerPath);
+        interProcessMessageSender.sendMessage(ExtensionMessage.OPEN_FILE_AT_LOCATION, [filename, lineNumber]);
+    }).catch(e => {
+        throw e;
```

Comment:
This catch is a no op. When you throw inside a promise/then/catch handler, that's equivalent to "reject" the promise chain or return Q.reject<Error>(e); This exception won't be catched by your sync catch